### ISSUE TITLE
Switch type assumes each cases value to be unique

### ISSUE THREAD

**Issue by bryanspears** [AmpersandJS/ampersand-dom-bindings#40]
For example:
```
type: 'switch',
cases: {
    'one': '#one',
    'two': '#one'',
    'three': '#two'
}
```

In the above example, if the value is `one`, `#one` will be hidden due to case `two` having the same target id.

**Comment by pgilad** [comment]
This is definitely an edge case - &-dom looks for a matching value and turns it visible, and then hides all the rest.
What is your purposed solution to this? If we have a match, before hiding the other elements should their value be compared to the matching element and then it should not be hidden?

I'm not sure users are actually expecting this behavior.
@AmpersandJS/core-team ?

See the failing test case I made: https://github.com/AmpersandJS/ampersand-dom-bindings/blob/failing-switch/test/index.js#L892-L928

**Comment by latentflip** [comment]
> This is definitely an edge case - &-dom looks for a matching value and turns it visible, and then hides all the rest.

I don't think that's quite correct. As per [these lines](https://github.com/AmpersandJS/ampersand-dom-bindings/blob/master/ampersand-dom-bindings.js#L199-L205) we're just iterating through the cases, showing and hiding as we go. I'm guessing from @bryanspears example and looking at the code, that if the value is `'one'` then `#one` will not be visible, because it'll be hidden when we test `'two'`. But if value is `'two'` `#one` will in fact be visible because of the order we iterate the object.

I don't think I'd be opposed to a PR which fixed this behaviour.

**Comment by pgilad** [comment]
I got...

**Comment by pgilad** [comment]
Will be fixed in my PR ^ #42

**Comment by pgilad** [comment]
Fixed by https://github.com/AmpersandJS/ampersand-dom-bindings/commit/e331ea2b6c9b0976ec11f272d05f121848caf70c
